bundles/org.openhab.binding.matter/matter-server/src/client/ControllerNode.ts

```diff
@@ -0,0 +1,367 @@
+// Include this first to auto-register Crypto, Network and Time Node.js implementations
+import { CommissioningController } from "@project-chip/matter.js";
+import { NodeId } from "@matter/types";
+import { PairedNode, NodeStates, Endpoint } from "@project-chip/matter.js/device";
+import { Environment, Logger, StorageContext } from "@matter/general";
+import { ControllerStore } from "@matter/node";
+import { WebSocketSession } from "../app";
+import { EventType, NodeState } from '../MessageTypes';
+import { printError } from "../util/error";
+const logger = Logger.get("ControllerNode");
+
+/**
+ * This class contains represents the Matter Controller / Admin client
+ */
+export class ControllerNode {
+
+    private environment: Environment = Environment.default;
+    private storageContext?: StorageContext;
+    private nodes: Map<NodeId, PairedNode> = new Map();
+    commissioningController?: CommissioningController;
+
+    constructor(
+        private readonly storageLocation: string,
+        private readonly controllerName: string,
+        private readonly nodeNum: number,
+        private readonly ws: WebSocketSession,
+        private readonly netInterface?: string
+    ) { }
+
+    get Store() {
+        if (!this.storageContext) {
+            throw new Error("Storage uninitialized");
+        }
+        return this.storageContext;
+    }
+
+    /**
+     * Closes the controller node
+     */
+    async close() {
+        await this.commissioningController?.close();
+        this.nodes.clear();
+    }
+
+    /**
+     * Initializes the controller node
+     */
+    async initialize() {
+        const outputDir = this.storageLocation;
+        const id = `${this.controllerName}-${this.nodeNum.toString()}`
+        const prefix = "openHAB: ";
+        const fabricLabel = prefix + this.controllerName.substring(0, 31 - prefix.length);
+                
+        logger.info(`Storage location: ${outputDir} (Directory)`);
+        this.environment.vars.set('storage.path', outputDir)
+    
+        //TODO we may need to choose which network interface to use
+        if (this.netInterface !== undefined) {
+            this.environment.vars.set("mdns.networkinterface", this.netInterface);
+        }
+        this.commissioningController = new CommissioningController({
+            environment: {
+                environment: this.environment,
+                id,
+            },
+            autoConnect: false,
+            adminFabricLabel: fabricLabel,
+        });
+        await this.commissioningController.initializeControllerStore();
+
+        const controllerStore = this.environment.get(ControllerStore);        
+        // TODO: Implement resetStorage
```

Comment:
Also don't forget to take care of this in a follow-up PR.

Reply:
Agreed, i have just struggled with the UI design part, would be good to get some ideas from others on that (maybe its a thing action?)

Reply:
I think having a Thing action is the most reasonably option here. Other than that there would be to add a command for that to the Karaf console, but that would provide far worse UX.